lightning/src/ln/onion_message.rs

```diff
@@ -0,0 +1,61 @@
+/// The payload of an inbound onion message.
+pub struct Payload {
+	/// Onion messages may contain custom TLVs for the recipient.
+	custom_tlvs: Vec<CustomTlv>,
+	// Coming soon: path_id, reply_path
+}
+
+/// A sender, receiver and forwarder of onion messages. In upcoming releases, this object will be
+/// used to retrieve invoices and fulfill invoice requests from offers.
+pub struct OnionMessager<Signer: Sign, K: Deref>
```

Comment:
Should this be `OnionMessager` or `InvoiceOnionMessager`?

Reply:
>  I guess its more of an intermediary struct that handles p2p onion messages, decrypts them, then hands them off to various other handler traits that actually handle the messages?

That'd make sense to me. For inbound OMs, we'd have an `InvoiceRequestHandler` (this would be another object that marshalls an invoice and gives it back to the `OnionMessager`) and an `InvoiceHandler` (this would probably be `ChannelManager` under the hood, to pay the invoice?) And a `CustomHandler` for custom onion messages. 

For outbound InvoiceRequests, `OnionMessager` could have a `request_invoice` method?

Reply:
Okay, so IIUC `OnionMessenger` will have a pointer to an `InvoiceHandler` that has a `send_payment` method that sends out an `invoice_request` and pays the invoice. I'm wondering, how does this `InvoiceHandler` communicate to the `OnionMessenger` when it has an invoice request it wants to send out?